src/main/java/com/products/demo/ProductController.java

```diff
@@ -0,0 +1,55 @@
+package com.products.demo;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Controller
+public class ProductController {
+
+    private ProductsRepository productsRepository;
+
+    public ProductController(ProductsRepository productsRepository) {
+        this.productsRepository = productsRepository;
+    }
+
+    @GetMapping("/")
+    public String getCategoryList(Model model) {
+
+        List<Category> categoryList = productsRepository.getAll().
+                stream().map(Product::getCategory).collect(Collectors.toList());
+        model.addAttribute("lista", categoryList);
```

Comment:
Na stronie głównej nie używasz tej listy. Dodatkowo pamiętaj, że "lista" to w tym przypadku nazwa zmiennej, której możesz używać później w szablonie - powinna być po angielsku